15-connect-app-and-store/README.md

```diff
@@ -0,0 +1,341 @@
+
+In order to actually connect the two, we will be using some helpers provided by [`react-redux`](https://github.com/reactjs/react-redux). Namely: [`<Provider/>`](https://github.com/reactjs/react-redux/blob/master/docs/api.md#provider-store) and [`connect()`](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options). The docs provide invaluable insight into their behavior, but in short terms:
+* **Provider**: The top level component in your application, makes the `store` accessible via calls to `connect()` for all children components
+* **connect()**: A function which takes your component as an argument, and and provides access to the `store` and `dispatch()` in the passed in component's props via [`mapStateToProps`](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) and [`mapDispatchToProps`](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options).
+
+A component which has been passed to connect takes on a different term: a [container](http://redux.js.org/docs/basics/UsageWithReact.html#presentational-and-container-components). The term comes from an abstraction suggested in redux documentation that components should be split into two basic groups: those which handle *presentation* and those which handle *data*. "Containers" are passed to `connect()` and so are aware of the state, and are able to dispatch actions, while presentational components simply consume their props and maintain state for UI purposes.
+
+In this step we are going to do all that is left in order to connect our React application to our Redux store and actions. This will include:
+* Creating our `store` in `App.js`
+* Applying `redux-thunk` as a middleware to handle async actions
+* Making `<Provider />` the top level component
+* Refactoring of `<App />` such that its primary job is creating the store, and hydrating the app. Moving the JSX into a new container component: `<Page />`
+* `connect()` `<Page />` so that we can `dispatch()` actions, and access our App's state
+
+The [tasks](#tasks) and [exercises](#exercises) below will address each of the steps above and when we finish, will have a fully functioning Redux app.
+
+As always, if you run into trouble with the [tasks](#tasks) or [exercises](#exercises), you can take a peek at the final [source code](src/).
+
+## Tasks
+
+First off Lets create a new folder [`src/containers/`](src/containers/) and within that folder add a new file [`Page.js`](src/containers/Page.js). Copy over the contents of `App.js` into the new file, and rename the class to `Page`:
```

Comment:
The CSS needs to be copied as well. Ideally the class names would switch to `page_*`